app/models/topic.rb

```diff
@@ -1040,7 +1040,10 @@ def invite(invited_by, username_or_email, group_ids = nil, custom_message = nil)
         raise UserExists.new(I18n.t("topic_invite.user_exists"))
       end
 
-      ensure_can_invite!(target_user, invited_by)
+      comm_screener = UserCommScreener.new(invited_by.id, target_usernames: target_user.username)
+      if comm_screener.ignoring_or_muting_actor?(target_user.id)
```

Comment:
I'm wondering why we need to pass in `target_user.username` during initialization and then `target_user.id` again when calling the `ignoring_or_muting_actor?` method.

Reply:
Ahh ok nevermind me. I read through the class and can see why we need to do this but I wonder if we can be consistent about whether the API works with the `User#id` or `User#username`.

Reply:
Good point, will change the methods like `ignoring_or_muting_actor?` to accept a username instead